### PR TITLE
Move configuration of tools to pyproject.toml

### PR DESCRIPTION
**Pull request by Secrus**
# Summary

# Checklist

- [x] I understand that this PR may be closed in case there was no previous discussion. (This doesn't apply to typos!)
- [**No changes needed**] I've added a test for each change that was introduced, and I tried as much as possible to make a single atomic change. 
- [**No changes needed**] I've updated the documentation accordingly.

I have noticed that the tools configured by `setup.cfg` file are the ones that can be configured in `pyproject.toml` which already held some of the tools configurations. Since `setup.cfg` is inherently connected with `setuptools` which is not used in this project. I didn't see a need for a discussion in this case.

### CODE REVIEW

**Review by Kludex — APPROVED** [review]

File: requirements.txt [hunk 1]
```diff
@@ -20,7 +20,7 @@ twine==4.0.2
 
 # Tests & Linting
 black==23.3.0
-coverage==7.2.2
+coverage[toml]==7.2.2
```
Comment (Kludex) [inline requirements.txt:23]: Why did you add this?

**Comment by Kludex** [comment]
Since we've removed flake8 (which was the only project holding this change), I think this is a good change. 👍

**Comment by Kludex** [comment]
Same PR welcome on other encode projects. 👀 🙏

**Comment by Secrus** [comment]
> Same PR welcome on other encode projects. eyes pray

Sure, give me a moment ;)

File: requirements.txt (hunk 1 above)
Comment (Secrus) [reply requirements.txt:23]: exactly. from what I have seen, coverage requires `toml` extra to work with `pyproject.toml` configs.
